api/integrations/github/github.py

```diff
@@ -16,14 +18,88 @@
     LINK_SEGMENT_TITLE,
     UNLINKED_FEATURE_TEXT,
     UPDATED_FEATURE_TEXT,
+    GitHubTag,
 )
 from integrations.github.dataclasses import GithubData
 from integrations.github.models import GithubConfiguration
 from integrations.github.tasks import call_github_app_webhook_for_feature_state
+from projects.tags.models import Tag, TagType
 from webhooks.webhooks import WebhookEventType
 
 logger = logging.getLogger(__name__)
 
+tag_by_event_type = {
+    "pull_request": {
+        "closed": GitHubTag.PR_CLOSED.value,
+        "converted_to_draft": GitHubTag.PR_DRAFT.value,
+        "opened": GitHubTag.PR_OPEN.value,
+        "reopened": GitHubTag.PR_OPEN.value,
+        "ready_for_review": GitHubTag.PR_OPEN.value,
+        "merged": GitHubTag.PR_MERGED.value,
+    },
+    "issues": {
+        "closed": GitHubTag.ISSUE_CLOSED.value,
+        "opened": GitHubTag.ISSUE_OPEN.value,
+        "reopened": GitHubTag.ISSUE_OPEN.value,
+    },
+}
+
+
+def handle_installation_deleted(payload: dict[str, typing.Any]) -> None:
+    installation_id = payload.get("installation", {}).get("id")
+    if installation_id:
+        try:
+            GithubConfiguration.objects.get(installation_id=installation_id).delete()
+        except GithubConfiguration.DoesNotExist:
+            logger.error(
+                f"Github Configuration with installation_id {installation_id} does not exist"
+            )
+
+
+def tag_feature_per_github_event(
+    event_type: str, action: str, metadata: dict[str, Any]
+) -> None:
+
+    # Get Feature with external resource of type GITHUB and url matching the resource URL
+    feature = Feature.objects.filter(
+        Q(external_resources__type="GITHUB_PR")
+        | Q(external_resources__type="GITHUB_ISSUE"),
+        external_resources__url=metadata.get("html_url"),
+    ).first()
+
+    if feature:
+        if (
```

Comment:
I would return early here. 

```python
if not feature:
    return
if ( #....
```

Reply:
In this case, it's the same, right? I will keep this since it is a positive condition, it is returning early anyway, and the code is readable in this way.